src/IO/Resources/AnimationsLoader.cs

```diff
@@ -1438,77 +1438,62 @@ private bool ReadUOPAnimationFrame(ushort animID, byte animGroup, byte direction
             animDirection.FrameCount = (byte)(type < ANIMATION_GROUPS_TYPE.EQUIPMENT ? Math.Round(fc / 5f) : 10);
             animDirection.SpriteInfos = new SpriteInfo[animDirection.FrameCount];
 
-            int headerSize = sizeof(UOPAnimationHeader);
-            int count = 0;
-
+            /* If the UOP packers weren't a bunch of idiots that randomly omit frames, we could
+             * just do this:
+             * reader.Skip(sizeof(UOPAnimationHeader) * direction * animDirection.FrameCount);
+             * but we can't. So we have to walk through the frames to seek to where we need to go.
+             */
             UOPAnimationHeader* animHeaderInfo = (UOPAnimationHeader*)reader.PositionAddress;
 
-            for (ushort i = 0, id = animHeaderInfo->FrameID, currentDir = 0; animHeaderInfo->FrameID < fc; ++i, ++id)
+            for (ushort currentDir = 0; currentDir <= direction; currentDir++)
             {
-                if (/*animHeaderInfo->FrameID != id*/ animHeaderInfo->FrameID - 1 == id || i >= animDirection.FrameCount)
+                for (ushort frameNum = 0; frameNum < animDirection.FrameCount; frameNum++)
                 {
-                    if (currentDir != direction)
+                    long start = reader.Position;
+                    animHeaderInfo = (UOPAnimationHeader*)reader.PositionAddress;
+
+                    if (animHeaderInfo->Group != animGroup || start + animHeaderInfo->DataOffset >= reader.Length)
                     {
-                        ++currentDir;
+                        throw new Exception("Corrupt UOP file!");
```

Comment:
UOP are imprevedible. I know. 
Instead of ruining the user experience just for an animation with an exception, we can go ahead i guess